ruby/tests/basic.rb

```diff
@@ -711,4 +711,15 @@ def test_oneof_fields_respond_to? # regression test for issue 9202
     assert_respond_to msg, :has_d?
     refute msg.has_d?
   end
+
+  def test_string_subclass
```

Comment:
This test doesn't exercise the same case as the description - `str` is a normal `String` and it doesn't look like any instances of `myString` are every created.

Reply:
Oops! I was wrong.
Fixed it in [7dad6cf](https://github.com/protocolbuffers/protobuf/pull/13818/commits/7dad6cfd187bceacc53625a6a8698aec97f1c14b)